sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -198,7 +198,7 @@ public void finish(
         if (!child.isFinished()) {
           child.setSpanFinishedCallback(
               null); // reset the callback, as we're already in the finish method
-          child.finish(SpanStatus.DEADLINE_EXCEEDED, finishTimestamp);
+          child.finish(child.getStatus(), finishTimestamp);
```

Comment:
hm, but as I understand, we don't even need this logic anymore at all (iterating through children and finishing them), relay will just do that for us?

Reply:
![image](https://github.com/getsentry/sentry-java/assets/4999776/caf13cb9-d2c4-4028-adc9-2374f3d0eef9)

Reply:
Oh right, good catch. Also: the status only gets set if there's no end timestamp: https://github.com/getsentry/relay/pull/1690/files#diff-7f49c53574a8b3eef93c44e6df72f927d591b4f00ebf791af696ea6eae77fb2bR280-R283
Let me fix that